shoplist/src/main/java/com/netguru/codereview/network/ShopListApiMock.kt

```diff
@@ -0,0 +1,42 @@
+package com.netguru.codereview.network
+
+import com.netguru.codereview.network.model.ShopListItemResponse
+import com.netguru.codereview.network.model.ShopListResponse
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+class ShopListApiMock : ShopListApi {
+    override suspend fun getShopLists(): List<ShopListResponse> =
+        coroutineScope {
+            List(9999) { index ->
+                ShopListResponse(
+                    list_id = index.toString(),
+                    userId = index,
+                    listName = "ListName$index"
+                )
+            }
+        }
+
+    override suspend fun getShopListItems(listId: String): List<ShopListItemResponse> =
+        coroutineScope {
+            Thread.sleep(2)
+            List(5) { index ->
+                ShopListItemResponse(
+                    itemId = index.toString(),
+                    name = "Name$index",
+                    quantity = 2.0
+                )
+            }
+        }
+
+    override fun getUpdateEvents(): Flow<String> = flow {
+        var counter = 0
+        while (true) {
```

Comment:
Thill will block the main thread if dispatcher context starts on Main Thread.
Try to change the context to IO thread for waiting process. We should avoid waiting with the Default or Main Thread.
Use withContext(Dispatcher.IO) to change the context to IO Thread as it is a light wait process and will not block the main thread.